doc/doxygen/src/debugging-aids.md

```diff
@@ -0,0 +1,32 @@
+# Debugging Tools {#debugging-tools}
+
+## Undefined Behavior Sanitizer (ubsan) {#ubsan}
+
+RIOT contains makefile convenience support for gcc/clang's undefined
+behaviour sanitizer.
+
+### Overview
+
+Both gcc and clang allow generation on code that does runtime checks for
+undefined behavior (UB).
+
+E.g., the following code might trigger UB for some parameters:
+
+```C
+void test(int foo) {
+    return (foo << 24);
+}
+```
+
+In this case, the signed shift would be alright unless it would "push out"
+bits to the left, with undefined runtime result. Using ubsan, this can be
+caught.
```

Comment:
```suggestion
In this case, the signed shift would be alright unless it would "push out" all
bits to the left, with undefined runtime result. Here, that happens on
architectures with 16-bit integers. Using ubsan, this can be caught.
```

Reply:
Actually, another undefined behavior is "foo is negative", see https://github.com/RIOT-OS/RIOT/pull/17278#discussion_r760015886. I'll also add that.